src/matchmaker.ts

```diff
@@ -212,6 +212,8 @@ export default class Matchmaker {
                         resolve(this.simBundle(paramsWithSignedTx, simOptions))
                     }
                 }
+                // manually call once in case tx is already landed
+                waitForTx()
```

Comment:
This is a very good idea. I wonder how it is implemented though. Because it is still going to subscribe because waitForTx isn't awaited, and we don't bail early. and if the code does match, we might removeListener before the listener is added (and then it will listen forever) . Maybe we could break out the code for submitting into a local function so waitForTx is small, and then we don't re-use it before starting listener, we just await checking for the tx directly and call that new function with the tx if it is present (and return before the on(block)) or proceed?